static/js/components/CourseEnrollmentDialog.js

```diff
@@ -99,10 +117,24 @@ export default class CourseEnrollmentDialog extends React.Component {
         </Button>
       )
     } else {
-      message = `You need to get a Personalized Course Price before
-        you can pay for this course. Or you can audit the course for FREE
-        and upgrade later. (Payment is required to get credit for the
-        MicroMasters certificate.)`
+      message = [
+        <p key="1">
+          You need to{" "}
+          <a
+            href="#"
+            className="calculate-link"
+            onClick={this.handleCalculatePriceClick}
```

Comment:
This is causing a `#` to be appended to the url, could you add a `e.preventDefault()` to the handler to prevent this?